common/src/main/java/com/wynntils/features/user/CustomNametagRendererFeature.java

```diff
@@ -7,33 +7,117 @@
 import com.wynntils.core.config.Config;
 import com.wynntils.core.features.UserFeature;
 import com.wynntils.core.managers.Model;
+import com.wynntils.core.webapi.WebManager;
+import com.wynntils.core.webapi.profiles.item.ItemProfile;
 import com.wynntils.mc.event.NametagRenderEvent;
+import com.wynntils.mc.utils.ComponentUtils;
+import com.wynntils.mc.utils.McUtils;
 import com.wynntils.wynn.model.UserInfoModel;
 import com.wynntils.wynn.objects.account.AccountType;
+import com.wynntils.wynn.utils.WynnPlayerUtils;
 import java.util.List;
+import net.minecraft.ChatFormatting;
+import net.minecraft.client.player.LocalPlayer;
+import net.minecraft.network.chat.TextComponent;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.entity.projectile.ProjectileUtil;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.Items;
+import net.minecraft.world.phys.AABB;
+import net.minecraft.world.phys.EntityHitResult;
+import net.minecraft.world.phys.Vec3;
 import net.minecraftforge.eventbus.api.SubscribeEvent;
 
 public class CustomNametagRendererFeature extends UserFeature {
+    private static final float RAYCAST_RANGE = 5f;
+
     @Config
     public boolean hideAllNametags = false;
 
     @Config
     public boolean showAccountType = true;
 
+    @Config
+    public boolean showGearOnHover = false;
+
     @SubscribeEvent
     public void onNameTagRender(NametagRenderEvent event) {
         if (hideAllNametags) {
             event.setCanceled(true);
             return;
         }
 
+        if (showGearOnHover) {
+            addGearNametag(event);
+        }
+
         if (showAccountType) {
-            for (AccountType type : AccountType.values()) {
-                if (type.getComponent() == null) continue;
+            addAccountTypeNametag(event);
+        }
+    }
+
+    private static void addGearNametag(NametagRenderEvent event) {
+        LocalPlayer player = McUtils.player();
+
+        Vec3 start = player.getEyePosition(1f);
+        Vec3 look = player.getLookAngle();
+        Vec3 direction = start.add(look.x * RAYCAST_RANGE, look.y * RAYCAST_RANGE, look.z * RAYCAST_RANGE);
+        AABB bb = player.getBoundingBox()
+                .expandTowards(look.x * RAYCAST_RANGE, look.y * RAYCAST_RANGE, look.z * RAYCAST_RANGE)
+                .expandTowards(1, 1, 1);
+
+        EntityHitResult hitResult = ProjectileUtil.getEntityHitResult(
+                McUtils.mc().level, player, start, direction, bb, (e) -> e instanceof Player);
+
+        if (hitResult == null) return;
+        if (hitResult.getEntity() != event.getEntity()) return;
+
+        Player hitPlayer = (Player) hitResult.getEntity();
+
+        if (hitPlayer.getScoreboardName().contains("§")) return; // npc
+        if (WynnPlayerUtils.isPlayerGhost(hitPlayer)) return;
```

Comment:
Combine these checks, also consider making a `WynnPlayerUtils#isNPC` as the `§` check has already been used elsewhere